docs/asciidoc/modules/ROOT/partials/usage/apoc.load.xml.adoc

```diff
@@ -277,6 +277,101 @@ RETURN author;
 | "Ralls, Kim"
 |===
 
+
+=== Avoid OOM using Xpath
+
+Generally, to avoid Heap Space Errors,
+to handle large files you should always try to return the result as a stream, and not as a unique result, to avoid `java.lang.OutOfMemoryError: Java heap space`, if possible
+For example, with a file like this:
+.book.xml
+[largeFile,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- <graphml xmlns="http://graphml.graphdrawing.org/xmlns" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://graphml.graphdrawing.org/xmlns http://graphml.graphdrawing.org/xmlns/1.0/graphml.xsd"> -->
+<graphml name="databases">
+<key id="name" for="node" attr.name="name"/>
+<key id="tagline" for="node" attr.name="tagline"/>
+<key id="title" for="node" attr.name="title"/>
+<key id="labels" for="node" attr.name="labels"/>
+<key id="summary" for="edge" attr.name="summary"/>
+<key id="label" for="edge" attr.name="label"/>
+<graph id="G" edgedefault="directed">
+  <node id="n0" labels=":Movie"><data key="labels">:Movie</data><data key="title">The Matrix</data><data key="tagline">Welcome to the Real World</data><data key="released">1999</data></node>
+  <node id="n1" labels=":Person"><data key="labels">:Person</data><data key="born">1964</data><data key="name">Keanu Reeves</data></node>
+  <node id="n2" labels=":Person"><data key="labels">:Person</data><data key="born">1967</data><data key="name">Carrie-Anne Moss</data></node>
+  <node id="n3" labels=":Person"><data key="labels">:Person</data><data key="born">1961</data><data key="name">Laurence Fishburne</data></node>
+  <node id="n4" labels=":Person"><data key="labels">:Person</data><data key="born">1960</data><data key="name">Hugo Weaving</data></node>
+  <node id="n5" labels=":Person"><data key="labels">:Person</data><data key="born">1967</data><data key="name">Lilly Wachowski</data></node>
+  <node id="n6" labels=":Person"><data key="labels">:Person</data><data key="born">1965</data><data key="name">Lana Wachowski</data></node>
+    // a lot of other node tags...
+
+  <edge id="e17" source="n3" target="n10" label="ACTED_IN"><data key="label">ACTED_IN</data><data key="roles">["Morpheus"]</data></edge>
+  <edge id="e18" source="n4" target="n10" label="ACTED_IN"><data key="label">ACTED_IN</data><data key="roles">["Agent Smith"]</data></edge>
+    // a lot of other edge tags...
+
+  <foo id="id2">foo2</foo>
+  <foo id="id3">foo3</foo>
+ // ...
+</graph>
+</graphml>
+
+----
+
+you can extract all the children of the `graph` tag via:
+
+[source,cypher]
+----
+CALL apoc.load.xml('largeFile.xml', '/graphml/*', {})
+----
+
+.Results
+[options="header"]
+|===
+| value
+| {"_children":[{"_type":"data","_text":":Movie","key":"labels"},{"_type":"data","_text":"The Matrix","key":"title"},{"_type":"data","_text":"Welcome to the Real World","key":"tagline"},{"_type":"data","_text":"1999","key":"released"}],"_type":"node","id":"n0","labels":":Movie"}
```

Comment:
This should be '/graphml/graph/*', shouldn't it?

Reply:
Also, I get slightly different results for my (outdated) 4.3 apoc version. I wonder if we might need different documentation there? :thought_balloon:

Reply:
So for clarification, the result I get from the first statement here is
`{"_type":"node","_children":[{"_type":"data","_text":":Movie","key":"l│
│abels"},{"_type":"data","_text":"The Matrix","key":"title"},{"_type":"│
│data","_text":"Welcome to the Real World","key":"tagline"},{"_type":"d│
│ata","_text":"1999","key":"released"}],"id":"n0","labels":":Movie"}`
So, I think the structure is inherently different. I neither updated my Neo4j version (4.3.2) nor apoc that I installed some time ago, so it might be because of that. But it's definitely interesting :smile:

Reply:
I see, yes it should depend on the version, thanks for the clarification